### PR TITLE
Opal 104 backend is able to refresh access tokens

### PR DESCRIPTION
**Pull request by arnikaeva**

### CODE REVIEW

**Review by bekobarna — DISMISSED** [review]

File: src/main/java/com/greenfoxacademy/kalendaryo/service/AuthAndUserService.java [hunk 1]
```diff
@@ -59,9 +60,12 @@ public String getRandomClientToken() {
     return Base64.encodeBase64String(random);
   }
 
-  private void saveGoogleAuth(GoogleAuth googleAuth) throws IOException {
-    String accessToken = authorization.authorize(googleAuth.getAuthCode());
+  protected void saveGoogleAuth(GoogleAuth googleAuth) throws IOException {
+    TokenResponse tokens = authorization.authorize(googleAuth.getAuthCode());
+    String accessToken = tokens.getAccessToken();
+    String refreshTOken = tokens.getRefreshToken();
```
Comment (bekobarna) [inline src/main/java/com/greenfoxacademy/kalendaryo/service/AuthAndUserService.java]: refreshT0ken?

File: src/main/java/com/greenfoxacademy/kalendaryo/service/KalendarService.java [hunk 2]
```diff
@@ -36,6 +41,7 @@
     public static final String GOOGLE_AUTH_NOT_FOUND = "GoogleAuth not found int database for the following email and ID=";
```
Comment (bekobarna) [inline src/main/java/com/greenfoxacademy/kalendaryo/service/KalendarService.java]: "GoogleAuth not found int database ... "  please correct it to "in database"

File: src/test/java/com/greenfoxacademy/kalendaryo/controllers/AuthControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            .andExpect(status().isOk());
+  }
+
+  @Test
+  public void postAuthShouldReturnHttp401WithoutClientToken() throws Exception {
+    mock.perform(post("/auth")
+            .contentType(contentType))
+            .andExpect(status().is4xxClientError());
+  }
+}
```
Comment (bekobarna) [inline src/test/java/com/greenfoxacademy/kalendaryo/controllers/AuthControllerTest.java]: empty line at the end

File: src/test/java/com/greenfoxacademy/kalendaryo/service/AuthAndUserServiceTest.java [hunk 4]
```diff
@@ -84,6 +77,15 @@ public void setKalUserForPostAuth_noToken(){
...
+    tokenResponse.setRefreshToken("RefreshToken");
+    when(authorization.authorize(anyString())).thenReturn(tokenResponse);
+
+    try {
+      assertTrue("returns Googleauth", authAndUserService.manageGoogleAuthForPostAuth(kalUser, googleAuth) != null);
+    } catch (NullPointerException ne) {
+      ne.getMessage();
+    }
   }
 }
```
Comment (bekobarna) [inline src/test/java/com/greenfoxacademy/kalendaryo/service/AuthAndUserServiceTest.java]: empty line at the end

File: src/main/java/com/greenfoxacademy/kalendaryo/service/authorization/AuthorizeKal.java [hunk 5]
```diff
@@ -30,6 +35,8 @@
 @Profile("dev")
 public class AuthorizeKal implements Authorization{
 
+    public static Integer FIRST_ATTEMPT = 1;
```
Comment (bekobarna) [inline src/main/java/com/greenfoxacademy/kalendaryo/service/authorization/AuthorizeKal.java]: we dont use this integer at all, do we need it?

**Review by bekobarna — APPROVED** [review]

**Review by FerencziSamu — APPROVED** [review]

**Pull request merged**